rubi/rubi/client.py

```diff
@@ -0,0 +1,635 @@
+import logging as log
+from _decimal import Decimal
+from multiprocessing import Queue
+from threading import Thread
+from time import sleep
+from typing import Union, List, Optional, Dict, Type, Any, Callable
+
+from eth_typing import ChecksumAddress
+from web3.types import EventData
+
+from rubi.contracts import (
+    RubiconMarket,
+    RubiconRouter,
+    ERC20,
+)
+from rubi.network import (
+    Network,
+)
+from rubi.types import (
+    OrderSide,
+    NewMarketOrder,
+    NewLimitOrder,
+    Pair,
+    OrderBook,
+    PairDoesNotExistException,
+    BaseEvent,
+    OrderEvent,
+    Transaction,
+    BaseNewOrder,
+    NewCancelOrder,
+    UpdateLimitOrder
+)
+
+
+class Client:
```

Comment:
i think in the future we will want to also include `swap` and `multiswap` functionality here. especially as the market starts to support more pairs, this will most likely be the easiest entry point to take advantage of price discrepancies across multiple books